cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAgentClient.java

```diff
@@ -0,0 +1,117 @@
+import io.strimzi.operator.cluster.model.DnsNameGenerator;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.Util;
+
+/**
+ * Creates HTTP client and interacts with Kafka Agent's REST endpoint
+ */
+class KafkaAgentClient {
+
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(KafkaAgentClient.class.getName());
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private static final String BROKER_STATE_REST_PATH = "/v1/broker-state/";
+    private static final int BROKER_STATE_HTTPS_PORT = 8443;
+
+    private String namespace;
+    private Reconciliation reconciliation;
+    private String cluster;
+    private Secret clusterCaCertSecret;
+    private Secret coKeySecret;
+
+    KafkaAgentClient(Reconciliation reconciliation, String cluster, String namespace, Secret clusterCaCertSecret, Secret coKeySecret) {
+        this.reconciliation = reconciliation;
+        this.cluster = cluster;
+        this.namespace = namespace;
+        this.clusterCaCertSecret = clusterCaCertSecret;
+        this.coKeySecret = coKeySecret;
+    }
+
+    protected String doGet(URI uri) {
+        try {
+            SSLContext sslContext = SSLContext.getInstance("TLSv1.3");
+            KeyStore trustStore = KeyStore.getInstance("PKCS12");
+            trustStore.load(new ByteArrayInputStream(
+                Util.decodeFromSecret(clusterCaCertSecret, "ca.p12")),
+                new String(Util.decodeFromSecret(clusterCaCertSecret, "ca.password"), StandardCharsets.UTF_8).toCharArray());
+
+            String trustManagerFactoryAlgorithm = TrustManagerFactory.getDefaultAlgorithm();
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(trustManagerFactoryAlgorithm);
+            trustManagerFactory.init(trustStore);
+            char[] keyPassword = new String(Util.decodeFromSecret(coKeySecret, "cluster-operator.password"), StandardCharsets.UTF_8).toCharArray();
+            KeyStore coKeyStore = KeyStore.getInstance("PKCS12");
+            coKeyStore.load(new ByteArrayInputStream(
+                Util.decodeFromSecret(coKeySecret, "cluster-operator.p12")),
+                keyPassword
+            );
+
+            String keyManagerFactoryAlgorithm = KeyManagerFactory.getDefaultAlgorithm();
+            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(keyManagerFactoryAlgorithm);
+            keyManagerFactory.init(coKeyStore, keyPassword);
+
+            sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
+
+
+            HttpRequest req = HttpRequest.newBuilder()
+                    .uri(uri)
+                    .GET()
+                    .build();
+
+            var response =  HttpClient.newBuilder()
+                    .sslContext(sslContext)
+                    .build()
```

Comment:
I think it would be more natural to configure the HttpClient once, initializing a field in the constructor. Then this `doGet` method just creates the request and handles the response. That would allow to reuse the same `HttpClient` between calls.